src/meltano/migrations/versions/13e8639c6d2b_add_state_edit_to_job_state_enum.py

```diff
@@ -37,7 +37,7 @@ def upgrade():
         op.alter_column(
             table_name="job",
             column_name="state",
-            type_=sa.types.String,
```

Comment:
@edgarrmondragon - I'm not an Alembic guru by any measure. From your experience, do you it okay to modify these already-created migrations?

Reply:
@aaronsteers So, if MySQL/MSSQL can't even run the existing migrations then this is only way to get them started. We should confirm that.

Something I'd worry about is migrations auto-generated going forward so I think we should also change the data types in the SQLAlchemy models themselves, which I've linked in https://github.com/meltano/meltano/pull/6167#issuecomment-1154592468

Reply:
> So, if MySQL/MSSQL can't even run the existing migrations then this is only way to get them started. We should confirm that.

@edgarrmondragon when i started going down the path of adding mysql support for the hackday that was definitely the issue I ran into. The initial migrations failed because MySQL needs a size constraint on varchar's (String() here)  

We *could* retro fit the migrations and branch based on dialect or driver or something along those lines. e.g. sqlite/postgres get the old style create tables while everything else gets create/alter tables with defined string sizes. If we went this route , I think opting sqlite/postgres into the old style is probably better than explicitly opting MySQL/MSSQL into the new style.

Reply:
@aaronsteers @edgarrmondragon Exactly, the problem is the fact that the alternate databases cannot run the existing migration. So creating an altering migration afterwards doesn't work in this case, as the migration cannot be reached.